xlml/utils/xpk.py

```diff
@@ -120,7 +130,10 @@ def wait_for_workload_completion(
     if pod.status.phase in ["Pending", "Running"]:
       logging.info(f"One pod phase is: {pod.status.phase}")
       return False
-    elif pod.status.phase in ["Failed", "Unknown"]:
+    elif pod.status.phase == "Failed":
+      # Don't keep retrying if the pod has failed
+      raise AirflowFailException(f"Bad pod phase: {pod.status.phase}")
```

Comment:
Does `AirflowFailException` prevent retries? That's good to know

Reply:
Yep! https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/exceptions/index.html#airflow.exceptions.AirflowFailException

We need this especially now that I've increased the XPK sensor timeout, otherwise it will keep querying the already-dead pods for the full timeout period.

Reply:
Nice!